helix-term/src/commands.rs

```diff
@@ -2243,14 +2244,33 @@ fn extend_search_prev(cx: &mut Context) {
 }
 
 fn search_selection(cx: &mut Context) {
+    search_selection_generic(cx, |selection, contents| {
+        regex::escape(&selection.fragment(contents))
+    })
+}
+
+fn search_selection_detect_word_boundaries(cx: &mut Context) {
+    search_selection_generic(cx, |selection, contents| {
```

Comment:
Instead of passing down a `impl Fn`, let's just pass a bool to a function like:

```rust
search_selection_impl(cx: &mut Context, detect_word_boundaries: bool)
```

and then we can replace the `map` with `regex::escape` with basically the `format!` in this block but use `if detect_word_boundaries && is_bow { "\\b" } else { "" }`.

Reply:
Done, assuming I understood your suggestion correctly 🙂